src/AspNet/WebApi/src/Asp.Versioning.WebApi/System.Net.Http/HttpResponseMessageExtensions.cs

```diff
@@ -35,6 +38,31 @@ public static void WriteSunsetPolicy( this HttpResponseMessage response, SunsetP
         AddLinkHeaders( headers, sunsetPolicy.Links );
     }
 
+    /// <summary>
+    /// Writes the sunset policy to the specified HTTP response.
+    /// </summary>
+    /// <param name="response">The <see cref="HttpResponseMessage">HTTP response</see> to write to.</param>
+    /// <param name="deprecationPolicy">The <see cref="DeprecationPolicy">deprecation policy</see> to write.</param>
+    public static void WriteDeprecationPolicy( this HttpResponseMessage response, DeprecationPolicy deprecationPolicy )
+    {
+        ArgumentNullException.ThrowIfNull( response );
+        ArgumentNullException.ThrowIfNull( deprecationPolicy );
+
+        var headers = response.Headers;
+
+        if ( deprecationPolicy.Date.HasValue )
```

Comment:
This can be pattern matched with:

```c#
if ( deprecationPolicy.Date is { } when )
```

FYI, this **cannot** be matched as:

```c#
if ( deprecationPolicy.Date is var when )
```

because `when` would be coerce to `DateTime?` whereas `{ }` does not allow `null`.